source/distributed/tria_base.cc

```diff
@@ -362,36 +362,125 @@ namespace parallel
+                for (auto j = 0u; j < GeometryInfo<dim>::vertices_per_face; ++j)
+                  {
+                    auto         v_t  = face_t->vertex_index(j);
+                    auto         v_n  = face_n->vertex_index(j);
+                    unsigned int temp = std::min(v_t, v_n);
+                    {
+                      auto it = vertex_to_coinciding_vertex.find(v_t);
+                      if (it != vertex_to_coinciding_vertex.end())
+                        temp = std::min(temp, it->second);
+                    }
+                    {
+                      auto it = vertex_to_coinciding_vertex.find(v_n);
+                      if (it != vertex_to_coinciding_vertex.end())
+                        temp = std::min(temp, it->second);
+                    }
+                    vertex_to_coinciding_vertex[v_t] = temp;
+                    vertex_to_coinciding_vertex[v_n] = temp;
+                  }
+              }
+
+      // 1b) compress map: let vertices point to the coinciding vertex with
+      //     the smallest index
+      for (auto &p : vertex_to_coinciding_vertex)
+        {
+          if (p.first == p.second)
+            continue;
+          unsigned int temp = p.second;
+          while (temp != vertex_to_coinciding_vertex[temp])
+            temp = vertex_to_coinciding_vertex[temp];
+          p.second = temp;
+        }
 
+#ifdef DEBUG
+      // check if map is actually compressed
+      for (auto p : vertex_to_coinciding_vertex)
+        {
+          if (p.first == p.second)
+            continue;
+          auto pp = vertex_to_coinciding_vertex.find(p.second);
+          if (pp->first == pp->second)
+            continue;
+          AssertThrow(false, ExcMessage("Map has to be compressed!"));
+        }
+#endif
 
-    std::vector<bool> vertex_of_own_cell(this->n_vertices(), false);
+      // 1c) create a map: smallest index of coinciding index -> all
+      // coinciding indices
+      std::map<unsigned int, std::set<unsigned int>>
+        smallest_coinciding_vertex_to_coinciding_vertices;
+      for (auto p : vertex_to_coinciding_vertex)
+        smallest_coinciding_vertex_to_coinciding_vertices[p.second] =
+          std::set<unsigned int>();
+
+      for (auto p : vertex_to_coinciding_vertex)
+        smallest_coinciding_vertex_to_coinciding_vertices[p.second].insert(
+          p.first);
+
+      // 1d) create a map: vertex -> all coinciding indices
+      for (auto &s : smallest_coinciding_vertex_to_coinciding_vertices)
+        for (auto &ss : s.second)
+          vertex_to_coinciding_vertices[ss] = s.second;
+    }
 
+    // 2) collect vertices belonging to local cells
+    std::vector<bool> vertex_of_own_cell(this->n_vertices(), false);
     for (const auto &cell : this->active_cell_iterators())
       if (cell->is_locally_owned())
-        for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_cell; ++v)
+        for (auto v = 0u; v < GeometryInfo<dim>::vertices_per_cell; ++v)
```

Comment:
Can you please undo this change? I find the specified type much easier to read than `0u`, especially for inexperienced users.